app/models/supplejack_api/search.rb

```diff
@@ -9,9 +9,16 @@ class Search
     def initialize(options = {})
       @original_options = options.dup
       klass = self.class
-      @options = SearchParams.new(
-        **options.merge(model_class: klass.model_class, schema_class: klass.schema_class)
-      )
+
+      @options = if options.present? && options['api_key'].present?
```

Comment:
I think it would be better to check the role of the key rather than if it is present or not, the anonymous user does actually have a key and it gets used if you don't provide your own key.